benchmarking/src/llmperf/sambanova_client.py

```diff
@@ -21,315 +23,503 @@
 
-    if request_config.mode == "stream":
-        path = f"/predict/generic/stream/{project_id}/{endpoint_id}"
-    else:
-        path = f"/predict/generic/{project_id}/{endpoint_id}"
+    def _populate_server_metrics(self, response_dict: dict, metrics: dict) -> dict:
+        """Parse output data to metrics dictionary structure
 
-    url = f"{base_url}{API_BASE_PATH}{path}"
-    return url
+        Args:
+            response_dict (dict): dict data with performance metrics
+            metrics (dict): metrics dictionary
 
+        Returns:
+            dict: updated metrics dictionary
+        """
 
-def _get_data(request_config: RequestConfig) -> dict:
-    """Gets data structure needed for API based on request_config
+        metrics[common_metrics.NUM_INPUT_TOKENS_SERVER] = response_dict.get(
+            "prompt_tokens_count"
+        ) or response_dict.get("prompt_tokens")
+        
+        metrics[common_metrics.NUM_OUTPUT_TOKENS_SERVER] = response_dict.get(
+            "completion_tokens_count"
+        ) or response_dict.get("completion_tokens")
+        
+        metrics[common_metrics.NUM_TOTAL_TOKENS_SERVER] = response_dict.get(
+            "total_tokens_count"
+        ) or response_dict.get("total_tokens")
+        ttft_server = response_dict.get("time_to_first_token") or response_dict.get(
+            "time_to_first_response"
+        )
+        
+        metrics[common_metrics.TTFT_SERVER] = ttft_server
+        
+        metrics[common_metrics.E2E_LAT_SERVER] = response_dict.get(
+            "total_latency"
+        ) or response_dict.get("model_execution_time")
+        
+        metrics[common_metrics.REQ_OUTPUT_THROUGHPUT_SERVER] = (
+            response_dict.get("completion_tokens_after_first_per_sec")
+            or response_dict.get("completion_tokens_per_sec_after_first_response")
+            or response_dict.get("throughput_after_first_token")
+        )
+        
+        metrics[common_metrics.TOTAL_TOKEN_THROUGHPUT_SERVER] = response_dict.get("total_tokens_per_sec") 
+        if (metrics[common_metrics.TOTAL_TOKEN_THROUGHPUT_SERVER] is None) and (metrics[common_metrics.E2E_LAT_SERVER] is not None):
+            metrics[common_metrics.TOTAL_TOKEN_THROUGHPUT_SERVER] = metrics[common_metrics.NUM_TOTAL_TOKENS_SERVER] / (metrics[common_metrics.E2E_LAT_SERVER])
+            
+        metrics[common_metrics.BATCH_SIZE_USED] = response_dict.get("batch_size_used")
+        
+        return metrics
 
-    Args:
-        request_config (RequestConfig): contains LLM params
+class SambaStudioAPI(BaseAPIEndpoint):
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        # Load sambastudio env variables
+        self.base_url = os.environ.get("SAMBASTUDIO_BASE_URL")
+        self.project_id = os.environ.get("SAMBASTUDIO_PROJECT_ID")
+        self.endpoint_id = os.environ.get("SAMBASTUDIO_ENDPOINT_ID")
+        self.api_key = os.environ.get("SAMBASTUDIO_API_KEY")
```

Comment:
why is base URI param not used here?

Reply:
that change will be included in the PR about Api v2. Please let's follow this on there